sdk/sync/subscription.go

```diff
@@ -64,27 +46,17 @@ func (s *subscription) process() {
 	conn := s.client.Conn()
 	defer conn.Close()
 
-	id, err := conn.ClientID().Result()
-	if err != nil {
-		s.connCh <- -1
-		s.result = fmt.Errorf("failed to get the current conn id: %w", err)
-		return
-	}
-
-	// store the conn ID in the channel.
-	s.connCh <- id
-
 	args := &redis.XReadArgs{
 		Streams: []string{key, "0"},
 		Block:   0,
 	}
 
 	var last redis.XMessage
-	for !s.isClosed() {
+	for {
```

Comment:
The connection _should_ obey the context that we've set on the client. We shouldn't need to play this close dance.

Reply:
It doesn't but I've fixed that.